### ISSUE TITLE
TableGateway used with TableIdentifier and MetadataFeature

### ISSUE THREAD

**Issue by jbelien** [zendframework/zend-db#391]
- [x] I was not able to find an [open](https://github.com/zendframework/zend-db/issues?q=is%3Aopen) or [closed](https://github.com/zendframework/zend-db/issues?q=is%3Aclosed) issue matching what I'm seeing.
 - [x] This is not a question. (Questions should be asked on [slack](https://zendframework.slack.com/) ([Signup for Slack here](https://zendframework-slack.herokuapp.com/)) or our [forums](https://discourse.zendframework.com/).)

Provide a narrative description of what you are trying to accomplish.

### Code to reproduce the issue

When using TableGateway with TableIdentifier and MetadataFeature, I have an error message (see below).
My database is *PostgreSQL 10.8*.

```php
$test = new TableGateway(new TableIdentifier('test', 'public'), $adapter, new MetadataFeature());
```

### Expected results

I'm expecting the same result that I get with the following code (that one works):

```php
$test = new TableGateway('test', $adapter, new MetadataFeature());
```

### Actual results

Error message:
```
ErrorException: Illegal offset type in isset or empty in file vendor/zendframework/zend-db/src/Metadata/Source/AbstractSource.php on line 449
```

See https://github.com/zendframework/zend-db/blob/master/src/Metadata/Source/AbstractSource.php#L449

**Comment by jbelien** [comment]
Seems to have been already mentioned in 2013 (https://github.com/zendframework/zendframework/issues/4287) but was never closed.

**Comment by jbelien** [comment]
I think I found the way to fix this ; see PR #392
